packages/@react-aria/color/src/useHexColorField.ts

```diff
@@ -0,0 +1,100 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaHexColorFieldProps} from '@react-types/color';
+import {Color, HexColorFieldState} from '@react-stately/color';
+import {HTMLAttributes, LabelHTMLAttributes, RefObject} from 'react';
+import {useSpinButton} from '@react-aria/spinbutton';
+import {useId} from '@react-aria/utils';
+import {useTextField} from '@react-aria/textfield';
+
+interface HexColorFieldAria {
+  labelProps: LabelHTMLAttributes<HTMLLabelElement>,
+  inputFieldProps: HTMLAttributes<HTMLInputElement>
+}
+
+export function useHexColorField(
+  props: AriaHexColorFieldProps,
+  state: HexColorFieldState,
+  ref: RefObject<HTMLInputElement>
+): HexColorFieldAria {
+  const {
+    value,        // eslint-disable-line @typescript-eslint/no-unused-vars
+    defaultValue, // eslint-disable-line @typescript-eslint/no-unused-vars
+    onChange,     // eslint-disable-line @typescript-eslint/no-unused-vars
+    ...otherProps
+  } = props;
+
+  const {
+    isDisabled,
+    isReadOnly,
+    isRequired,
+    minValue = '#000000',
+    maxValue = '#ffffff',
+  } = otherProps;
+  
+  const {
+    colorValue,
+    inputValue,
+    increment,
+    decrement,
+    incrementToMax,
+    decrementToMin,
+  } = state;
+
+  const minColor = typeof minValue === 'string' ? new Color(minValue) : minValue;
+  const minColorString = minColor.toString('hex').substring(1);
+  const minColorNumber = parseInt(minColorString, 16);
+
+  const maxColor = typeof maxValue === 'string' ? new Color(maxValue) : maxValue;
+  const maxColorString = maxColor.toString('hex').substring(1);
+  const maxColorNumber = parseInt(maxColorString, 16);
+
+  const colorValueString = colorValue.toString('hex').substring(1);
+  const colorValueNumber = parseInt(colorValueString, 16);
+
+  const {spinButtonProps} = useSpinButton(
+    {
+      isDisabled,
+      isReadOnly,
+      isRequired,
+      maxValue: maxColorNumber,
+      minValue: minColorNumber,
+      onIncrement: increment,
+      onIncrementToMax: incrementToMax,
+      onDecrement: decrement,
+      onDecrementToMin: decrementToMin,
+      value: colorValueNumber,
+      textValue: inputValue
+    }
+  );
+
+  const inputId = useId();
+  let {labelProps, inputProps} = useTextField({
+    ...otherProps,
+    id: inputId,
+    value: inputValue,
+    type: 'text',
+    autoComplete: 'off',
+    onChange: (val) => console.log(val)
+  }, ref);
+  
+  return {
+    labelProps,
+    inputFieldProps: {
+      ...inputProps,
+      onKeyDown: spinButtonProps.onKeyDown,
+      onFocus: spinButtonProps.onFocus,
+      onBlur: spinButtonProps.onBlur
```

Comment:
I'm not sure if we need to adopt all props coming in from `spinButtonProps`. For example, if we do, `aria-maxvalue` and `aria-minvalue` would be number/decimal values of the hex strings.

<img width="470" alt="Screen Shot 2020-09-04 at 12 13 32 PM" src="https://user-images.githubusercontent.com/5743833/92277642-42272700-eea8-11ea-83ae-411e376e10da.png">

Reply:
Add `role `even when there's no physical buttons in the UI

yes valuenow valuetext
no valuemin valuemax

Reply:
@majornista What should the values be for say #ff88a0?
aria-valuenow: '#ff88a0' || 0xff88a0 || 16746656
aria-valuetext: '#ff88a0'??
aria-valuemin: '#000000' || 0x000000 || 0
aria-valuemax: '#FFFFFF' || 0xFFFFFF || 16777215

my understanding is that value(min/max) must be a number, so the second two options would be best?
value text though, is it just the hex string or should it be something else entirely?

Reply:
`aria-valuenow`, `aria-valuemin` and `aria-valuemax` must be a number:

- `aria-valuenow`: 0xff88a0 || 16746656
- `aria-valuemin`: 0x000000 || 0
- `aria-valuemax`: 0xFFFFFF || 16777215

`aria-valuetext` is a string, so for the time being it can just be the hex string that appears as the text value for the input.

If we get into trying to better express the color using localized strings for accessibility, we can overload the `aria-valuetext` property to read as something like "#ff88a0, pale pink-red". See https://www.beautycolorcode.com/ for an example of color descriptions associated with hex colors.

Reply:
thanks for the clarification
we won't worry about the color descriptions for the time being

Reply:
Resolved https://github.com/adobe/react-spectrum/pull/986#discussion_r493979668